indico_toolkit/indico_wrapper/workflow.py

```diff
@@ -156,23 +149,41 @@ def update_workflow_settings(
             )
         )
 
-    def wait_for_submission_status_complete(
+    def wait_for_submission(
         self,
         submission_id: int,
-        seconds_to_sleep_each_loop: int = 2,
```

Comment:
I think you would be ok keeping this delay kwarg. If these are a lot of large docs you might expect this call to take a while.  You may want to reduce the # of requests in this case by increasing the sleep time.

Another option which is used in the SDK is the Debouncer which sleeps for an increasing amount of time in between calls.

Reply:
using built in